sale_project_timesheet_by_seniority/models/account_analytic_line.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = 'account.analytic.line'
+
+    @api.model
+    def _update_project_soline_mapping(self, vals):
```

Comment:
Take a look on https://github.com/OCA/timesheet/pull/242 - it may be easier to implement that way